packages/config/src/options.js

```diff
@@ -0,0 +1,39 @@
+const { cwd: getCwd } = require('process')
+const { dirname } = require('path')
+
+const findUp = require('find-up')
+
+// Normalize options and assign default values
+const normalizeOpts = async function(opts = {}) {
+  const optsA = { ...DEFAULT_OPTS, ...opts }
+  const optsB = await addDefaultRepositoryRoot(optsA)
+  return optsB
+}
+
+const DEFAULT_OPTS = {
+  cwd: getCwd(),
+}
+
+// Add a default value for the `repositoryRoot` option.
+// Do it by trying to find a `.git` directory up from `cwd`.
+// If nothing is found, default to `cwd`.
+const addDefaultRepositoryRoot = async function(opts) {
+  const repositoryRoot = await getRepositoryRoot(opts)
+  return { ...opts, repositoryRoot }
+}
+
+const getRepositoryRoot = async function({ repositoryRoot, cwd }) {
+  if (repositoryRoot !== undefined) {
+    return repositoryRoot
+  }
+
+  const repositoryRootA = await findUp('.git', { cwd, type: 'directory' })
+
+  if (repositoryRootA === undefined) {
+    return cwd
+  }
+
+  return dirname(repositoryRootA)
```

Comment:
Nitpicking: the A/B naming convention is non-descriptive. Not a blocker though.

Reply:
Yes this is through the whole codebase, so this might be worth discussing.

I use this convention when the variable new value does not change its meaning. For example, when a default value is assigned. Usually users would do:

```js
let myVariable = getMyVariable()
myVariable = myVariable === undefined ? true : myVariable 
```

But the codebase tries to follow functional programming principles: avoiding state mutation, enforcing immutability and creating new variables instead. IMHO state mutation makes it harder to understand code without context, as one needs to read all the statements where the variables is being mutated.

This translates into `const` being used instead of `var`/`let`:

```js
const myVariable = getMyVariable()
const myVariableWithDefault = myVariable === undefined ? true : myVariable 
```

Sometimes the value of `myVariable` will be changed several times in a function, but its meaning not so much. So this creates several new names for this variable (such as `myVariableWithDefault`) which feels a little artificial. Coming up with those names takes few seconds, and they tend to be longer to type.

Instead we can use an incremental naming convention like `myVariable1`, `myVariable2`, etc. or `myVariableA`, `myVariableB`, etc.

```js
const myVariable = getMyVariable()
const myVariableA = myVariable === undefined ? true : myVariable 
```